app/service/rest_svc.py

```diff
@@ -543,41 +546,41 @@ async def _persist_ability(self, access, ab):
         ab['tactic'] = ab.get('tactic').lower()
 
         # Validate platforms, ability will not be loaded if empty
-        if not ab.get('platforms'):
-            self.log.debug('At least one platform/executor is required to save ability.')
+        if not ab.get('platforms', dict()):
+            self.log.debug('At least one executor is required to save ability.')
             return []
 
-        # Create a new ability to be used for updating/creating
-        new_ability, new_ability_exec_timeouts = await self._prep_new_ability(ab)
-
         # Update or create ability
         _, file_path = await self.get_service('file_svc').find_file_path('{}.yml'.format(ab['id']), location='data')
         if file_path:
             # Ability exists, update
             current_ability = dict(self.strip_yml(file_path)[0][0])
             current_ability, current_parsers = await self._strip_parsers_from_ability(current_ability)
-            current_ability.update(new_ability)
+            current_ability.update(ab)
             final = await self._add_parsers_to_ability(current_ability, current_parsers)
             # Get access
             found_abilities = await self.get_service('data_svc').locate('abilities', dict(ability_id=ab['id']))
             allowed = found_abilities[0].access if found_abilities else self._get_allowed_from_access(access)
         else:
             # Create new ability, create file / directory
-            tactic_dir = os.path.join('data', 'abilities', new_ability.get('tactic'))
+            tactic_dir = os.path.join('data', 'abilities', ab.get('tactic'))
             if not os.path.exists(tactic_dir):
                 os.makedirs(tactic_dir)
-            file_path = os.path.join(tactic_dir, '%s.yml' % new_ability['id'])
-            final = new_ability
+            file_path = os.path.join(tactic_dir, '%s.yml' % ab['id'])
+            final = ab
             # Get access
             allowed = self._get_allowed_from_access(access)
 
         await self.get_service('file_svc').save_file(file_path, yaml.dump([final], encoding='utf-8', sort_keys=False),
                                                      '', encrypt=False)
         await self.get_service('data_svc').remove('abilities', dict(ability_id=final['id']))
         await self.get_service('data_svc').load_ability_file(file_path, allowed)
-        await self._restore_exec_timeouts(final['id'], new_ability_exec_timeouts)
-        return [a.display for a in
-                await self.get_service('data_svc').locate('abilities', dict(ability_id=final['id']))]
+
+        loaded_abilities = await self.get_service('data_svc').locate('abilities', dict(ability_id=final['id']))
+        if not loaded_abilities:
+            self.log.error('An error occurred when attempting to load the newly-created ability')
+            return {}
```

Comment:
Should this be an empty list?

Reply:
I had changed `r.extend()` to `r.append()` in `persist_ability()` and these changes have `_persist_ability()` returning a single ability, rather than a list of abilities with unique platforms/executors. Just updated the documentation to reflect this.